Merging the Identity database with our Database

XanderVeldeman opened the pull request:
# We're in the Endgame now
- Merged the databases into the `TheSpanishDatabase`
- Removed the UserManager => use the Identity UserManager **OR** the UserService
- UserID is now TEXT/STRING instead of INT
- We now use the Domain.Identity.UIMVCUser (I know I'm great at naming things) instead of the regular User
- User and UserDetails are still available in the domain BUT WILL NOT BE USED
- DTO's are now public (Identity needs to be able to directly communicate with the Context)

Next item will be rewriting the Create and Insert scripts.
![hulkbuster](https://media.giphy.com/media/Tzbksv6yHCXjW/giphy.gif)

XanderVeldeman committed:
Merged our DB with the Identity DB

* still need to link them with oneanother (FK's)
* we now only use one DBContext
* TheSpanishDatabase is the one true Database

XanderVeldeman committed:
Merged Identity and our database

* All userdata is found in the table AspNetUsers
* Userdata must be requested by the identity usermanager
* cucked all of the things you're currently working on
* modified the database so that it uses the new data
* modified all references userID from int to string
* modified all references of User to UIMVCUser

XanderVeldeman committed:
Merge branch 'dev' into Identity+TSI

XanderVeldeman committed:
Fixes after merge

It should be working now, I hope

XanderVeldeman committed:
Merge branch 'dev' into Identity+TSI

XanderVeldeman committed:
Fixes after pull with dev

XanderVeldeman committed:
Merge branch 'dev' into Identity+TSI

XanderVeldeman committed:
Fixes for merge

* Fixed the merge conflict issues

NielsSensei commented:
Note to all: Let me have an inquisition on this branch

NielsSensei commented on DAL/Data Transfer Objects/UserActivitiesDTO.cs:
```diff
@@ -4,7 +4,7 @@
+    public class UserActivitiesDTO
     {
         public int ActivityID { get; set; }
         public int UserID { get; set; }
```
UserID is nog altijd een int

XanderVeldeman replied:
Removed

XanderVeldeman replied:
Kheb ook de DB aangepast

NielsSensei commented on DAL/Data Transfer Objects/UserDetailsDTO.cs:
```diff
@@ -4,7 +4,7 @@
 namespace DAL.Data_Transfer_Objects
 {
-    class UserDetailsDTO
+    public class UserDetailsDTO
```
Volgens de PR is Users en UserDetails ongebruikt, waarom blijven ze dan bestaan.

XanderVeldeman replied:
Removed

NielsSensei commented on DAL/Data Transfer Objects/UsersDTO.cs:
```diff
@@ -4,7 +4,7 @@
 namespace DAL.Data_Transfer_Objects
 {
-    class UsersDTO
+    public class UsersDTO
```
Volgens de PR is Users en UserDetails ongebruikt, waarom blijven ze dan bestaan.

XanderVeldeman replied:
Removed

NielsSensei committed:
Ik denk dat alle errors in orde zijn.

NielsSensei reviewed (changes requested):
Goed idee en al maar ik was er vanaf het begin al tegen, test alles eens opnieuw na mijn 'fixes'

Ik denk als het echt niet gaat bij iemand anders dan bij ou dat ge gaat moeten werken met 2 dbContexten.

No Other choice.

First Deny
![deny](https://media.tenor.com/images/bf1f99c53272eedd23dc0b734ddce8f2/tenor.gif)

Code Knight Out
![knight](https://media.tenor.com/images/df2cc7d8d5e76e1dd460adce9546e843/tenor.gif)

XanderVeldeman committed:
Fixed the 'fixes'

- removed the User and UserDetails DTO's
- fixed your 'fixes'

NielsSensei committed:
Adding the missing Package

PackageManager ftw

XanderVeldeman committed:
fuck me and fuck the Rider tooltips

Probleem was dus blijkbaar dat de identity stores werden geaccessed via een hintpath die de Rider tooltip had toegevoegd ipv een package reference.

XanderVeldeman commented:
Nu zou™ het moeten werken.

NielsSensei reviewed (approved):
De Meeste features werken zoals gewoonlijk (kzal de branch open houden voor experimentatie if ya want): Ik ga wachten tot de merge totdat ik hierover reactie krijg.

Nog TODO's:
Alle tabellen terug deftig geseed.
UserService.cs nog herconfigureren (bvb. probeer naar een Detailview van een idee te gaan. Dat soort features werken nimeer)
Ik kwam het accidenteel tegen. Als ge probeert in te loggen met false credentials wordt dit niet opgevangen atm en krijg je een InvalidOperationException omdat hij je zoekt tussen loginProviders en niet vindt obviously.

![smollknight](https://media.tenor.com/images/3a8865f141fffa27526c5c4f083a0c13/tenor.gif)

XanderVeldeman committed:
Quickfix for UserService/CollectUserName

If it can't find a user it will return a string stating NOTFOUND

XanderVeldeman committed:
Fixed google login
